### PR TITLE
Sulaiman week 2

### PR DESCRIPTION
**Pull request by SULAIMAN-100**
# Your Details

Your Name:
Your City:
Your Slack Name:

# Homework Details

Module:
Week:

### CODE REVIEW

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 1]
```diff
@@ -11,9 +20,13 @@ function mood() {
   }
 }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/1-fix-functions.js:21]: What you did here is correct but you have removed the else statement inside the mood function. It should be there if the isHappy is false the else should be "I'm not happy" and true for the if "i'm happy".Fix that Sulaiman.

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 2]
```diff
@@ -11,9 +20,13 @@ function mood() {
...
 
+// 2) For any numerical value greater or equal to 10
+//    Hint: use constant `num` and only change isBigEnough.
+//    Variable isBigEnough needs to evaluate to a boolean
+
 function greaterThan10() {
-  let num = 10;
-  let isBigEnough;
+  const num = 10;
+  const isBigEnough = num >=num ;
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/1-fix-functions.js:29]: Here Sulaiman this variable should check if num is greater than a number that you choose and use this variable in the if/else statement to get the appropriate answer. It needs to be fixed.

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 3]
```diff
@@ -22,25 +35,39 @@ function greaterThan10() {
   }
 }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/1-fix-functions.js:36]: You have deleted the else statement here

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 4]
```diff
@@ -22,25 +35,39 @@ function greaterThan10() {
...
+//    Remember to Google how to use sort method
+
 function sortArray() {
-  let letters = ["a", "n", "c", "e", "z", "f"];
-  let sortedLetters;
+  const letters = ["a", "n", "c", "e", "z", "f"];
+  let sortedLetters = letters.sort();
 
   return sortedLetters;
 }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/1-fix-functions.js:47]: Well done here

File: week-2/Homework/mandatory/1-fix-functions.js [hunk 5]
```diff
@@ -22,25 +35,39 @@ function greaterThan10() {
...
+//    For example with array `[1, 2, 3, 4, 5]` it needs to return `4`
+//
+//    Hint: remember that arrays are zero-index based
+
+function get3rdIndex(arr) { 
+  let element = arr[3];
 
   return element;
+
 }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/1-fix-functions.js:71]: Good

File: week-2/Homework/mandatory/2-function-creation.js [hunk 6]
```diff
@@ -5,7 +5,13 @@ Write a function that:
...
 - makes the string all lowercase
 */
-function tidyUpString(strArr) {}
+function tidy(name) {
+  return name.trim().toLowerCase().replace("/", "");
+}
+ function tidyUpString(strArr) {
+  var tidystr = strArr.map(tidy);
+ return tidystr
+ }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/2-function-creation.js:14]: Sulaiman actually you are passing an array as a parameter. You can go through an array by using for loop. Map is used to create a new array with some modification of the first array. You can either use for loop or map to solve this exercise. Try to do it again.

File: week-2/Homework/mandatory/2-function-creation.js [hunk 7]
```diff
@@ -15,7 +21,13 @@ Complete the function to check if the variable `num` satisfies the following req
...
 */
 
-function validate(num) {}
+function validate(num) {
+  if (num===Number || num % 2 ===0 && num<=100){
+    return true
+  } else { 
+    return false
+  }
+}
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/2-function-creation.js:30]: In this function, you forgot to add typeOf num. the rest is correct

File: week-2/Homework/mandatory/2-function-creation.js [hunk 8]
```diff
@@ -26,7 +38,8 @@ The function must:
 */
 
 function remove(arr, index) {
-  return; // complete this statement
+arr.splice(index, 1);
+  return arr;
 }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/2-function-creation.js:43]: This is correct, for best practice store the parameter in a variable then apply the splice method to that variable.

File: week-2/Homework/mandatory/2-function-creation.js [hunk 9]
```diff
@@ -37,10 +50,16 @@ Write a function that:
...
-  
+function formatPercentage(arr) { 
+   for (let i=0; i<  arr.length; i++) {
+    if (arr[i] > 100){
+       arr[i] = 100
+      }
+       arr[i]= Number.parseFloat(arr[i].toFixed(2))+ "%";
+   }
+  return arr  ;
 }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/2-function-creation.js:61]: Perfect.

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 10]
```diff
@@ -7,31 +13,44 @@
...
   2. Remove the line that throws the error.
   3. What is printed to the console?
+            2     6    4    9   6    13   8
+      /// 2 the value of x , 6 the value of a , moving inside 'for loop' where i starts with value 0 and end with 4,
+     console d must be an even number and console e must be an odd number
+     first console d prints  4 where i=0 + x=2 + x=2 based on the f2 called function, 
+    then console e = 9 where i=1 + a=6 + the increment of a=2,,,,, back to the loop 6 is printed as value
+    of d where i=2 + x=2 + x=2,, console 13 is e when i=3 + a=6 + the increment a=4,,,,, last printed console is 8 
+      when i=4 + x=2 + x=2.
+
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/3-playing-computer.js:28]: you have been ask to tell what the out of each number belongs to. for example x = 2. Could fix this please

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 11]
```diff
@@ -7,31 +13,44 @@
...
   4. How many times is "f1" called?
+    ///// 2 times when i 1 and 3 
   5. How many times is "f2" called?
+  ////// 3 times when i was 0, 2, 4 
   6. What value does the "a" parameter take in the first "f1" call?
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/3-playing-computer.js:33]: you haven't answered this question

File: week-2/Homework/mandatory/3-playing-computer.js [hunk 12]
```diff
@@ -7,31 +13,44 @@
...
   7. What is the value of the "a" outer variable when "f1" is called for the first time?
+       6 + 2 the increment 'a = a + 1 ;' = 8
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/3-playing-computer.js:35]: This answer is wrong it is 6

File: week-2/Homework/mandatory/4-sorting-algorithm.js [hunk 13]
```diff
@@ -6,16 +12,35 @@ Today, you will be applying the sorting algorithm you used in that exercise in c
...
+    for (let i=0; i < sortedArr.length; i++){
+      for (let j=0; j < sortedArr.length; j++){
+        if (sortedArr[j] > sortedArr[j + 1]){
+           array = sortedArr[j]
+           sortedArr[j]= sortedArr[j+1]
+           sortedArr[j+1]= array
+        }
+      }
+    
+    }
```
Comment (Abdoulrazack95) [inline week-2/Homework/mandatory/4-sorting-algorithm.js:39]: Great job with the sort algorithm

File: week-2/Homework/extra/1-radio-stations.js [hunk 14]
```diff
@@ -26,7 +26,6 @@
  */
```
Comment (Abdoulrazack95) [inline week-2/Homework/extra/1-radio-stations.js:26]: There is no function here. Could you fix it

File: week-2/Homework/extra/2-array-methods.js [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+*/
+//1)Find the Smallest Number in an Array
+//Create a function that takes an array of numbers and return the smallest number in it
+
+function findSmallestNum(arr) {}
+
+findSmallestNum([354, 15, 114, 2]); //Expected: 2
+
+findSmallestNum([-76, 1, -79, 1, 0]); //Expected: -79
```
Comment (Abdoulrazack95) [inline week-2/Homework/extra/2-array-methods.js:12]: no work here

File: week-2/Homework/extra/2-array-methods.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+//4)Add the index to the number
+//Given an array of numbers, create a function which returns the same array but with each element's index in the array added to itself. This means you add 0 to the number at index 0, add 1 to the number at index 1, etc...
+function addIndexes(arr) {}
+addIndexes([1, 4, 3, 4, 5]); //Expected: [1, 5, 5, 7, 9]
+
+//4)Get the Sum of All Array Elements
+//Create a function that takes an array and returns the sum of all numbers in the array.
+function getSumOfItems(arr) {}
+
+getSumOfItems([-2, 84, 23]); //Expected: 105
```
Comment (Abdoulrazack95) [inline week-2/Homework/extra/2-array-methods.js:35]: no Work on this exercise

File: week-2/Homework/mandatory/1-fix-functions.js (hunk 1 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/1-fix-functions.js:21]: I did not remove the else statement  ,,, it is inside ... I will only add ===  true inside if statement

File: week-2/Homework/mandatory/1-fix-functions.js (hunk 2 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/1-fix-functions.js:29]: got it

File: week-2/Homework/mandatory/1-fix-functions.js (hunk 3 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/1-fix-functions.js:36]: it is there

File: week-2/Homework/mandatory/2-function-creation.js (hunk 6 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/2-function-creation.js:14]: done

File: week-2/Homework/mandatory/2-function-creation.js (hunk 7 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/2-function-creation.js:30]: done

File: week-2/Homework/mandatory/2-function-creation.js (hunk 8 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/2-function-creation.js:43]: done

File: week-2/Homework/mandatory/3-playing-computer.js (hunk 10 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/3-playing-computer.js:28]: done

File: week-2/Homework/mandatory/3-playing-computer.js (hunk 11 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/3-playing-computer.js:33]: answered

File: week-2/Homework/mandatory/3-playing-computer.js (hunk 12 above)
Comment (SULAIMAN-100) [reply week-2/Homework/mandatory/3-playing-computer.js:35]: edited

**Pull request closed without merging**